lib/trento.pm

```diff
@@ -0,0 +1,300 @@
+                    script_run($kubectl_logs_cmd, 180);
+                    upload_logs($log_txt);
+                }
+            }
+        }
+        script_run('ls -lai *.txt', 180);
+    }
+}
+
+
+=head2 podman_self_check
+Perform some generic checks realted to the podman installation itself
+and the relevant paramenters of the current machine environment.
+
+=cut
+sub podman_self_check {
+    record_info('PODMAN', 'Check podman');
+    assert_script_run('which podman');
+    assert_script_run('podman --version');
+    assert_script_run('podman info --debug');
+    assert_script_run('podman ps');
+    assert_script_run('podman images');
+    assert_script_run('df -h');
+}
+
+
+use constant PODMAN_PULL_LOG => '/tmp/podman_pull.log';
+
+
+=head2 crypress_install_container
+Prepare whatever is needed to run cypress tests using container
+
+=cut
+sub cypress_install_container {
+    my ($cypress_ver) = @_;
+    podman_self_check();
+
+    # list all the available cypress images
+    my $cypress_image = 'docker.io/cypress/included';
+    assert_script_run('podman search --list-tags ' . $cypress_image);
+
+    # Pull in advance the cypress container
+    my $podman_pull_cmd = 'time podman ' .
+      '--log-level trace ' .
+      'pull ' .
+      '--quiet ' .
+      $cypress_image . ':' . $cypress_ver .
+      ' | tee ' . PODMAN_PULL_LOG;
+    assert_script_run($podman_pull_cmd, 1800);
+    assert_script_run('df -h');
+    assert_script_run('podman images');
+}
+
+
+use constant CYPRESS_LOG_DIR => '/root/result';
+
+
+=head2 cypress_log_upload
+Upload to openQA the relevant logs
+Optional argument array of strings with relevant file extensions
+(dot needed)
+
+=cut
+sub cypress_log_upload {
+    my $self = shift;
+    my @log_filter = @_;
+    my $find_cmd = 'find ' . CYPRESS_LOG_DIR . ' -type f \( -iname \*' . join(' -o -iname \*', @log_filter) . ' \)';
+
+    upload_logs("$_") for split(/\n/, script_output($find_cmd));
+}
+
+
+=head2 cypress_exec
+Execute a cypress command within the container 
+
+
+=cut
+sub cypress_exec {
+    my ($self, $cypress_test_dir, $cmd, $timeout, $log_prefix, $failok) = @_;
+    my $ret = 0;
+
+    record_info('CY EXEC', 'Cypress exec:' . $cmd);
+    my $cypress_ver = get_var('TRENTO_CYPRESS_VERSION', '4.4.0');
+    my $container_name = "docker.io/cypress/included:$cypress_ver";
+    # container is executed with --name to easy the log retrieve
+    # to do so we need to rm eventually present container with that name
+    script_run('podman rm goofy || echo "No Goofy to delete"');
+    my $cypress_run_cmd = 'podman run ' .
+      '-it --name goofy ' .
+      "-v " . CYPRESS_LOG_DIR . ":/results " .
+      "-v $cypress_test_dir:/e2e -w /e2e " .
+      '-e "DEBUG=cypress:*" ' .
+      '--entrypoint=\'[' .
+      '"/bin/sh", "-c", ' .
+      ' "/usr/local/bin/cypress ' . $cmd .
+      ' 2>/results/cypress_' . $log_prefix . '_log.txt"' .
+      ']\' ' . $container_name .
```

Comment:
Be careful with the terminology. If names in podman are the same as in docker, then `$container_name` is actually the **Image Name** and what you are passing to the command via `--name` is the **Container Name**. (I could be wrong though ... haven't used podman yet)